contracts/V1/DisputeTypes.sol

```diff
@@ -110,17 +110,22 @@ enum ProofType {
 // ========================== Block related fraud proofs ==========================
 struct BlockEmptyProof {
     SignedBlock emptyBlock;
+    StateSnapshot latestStateSnapshot;
+    bytes32 previousStateSnapshotHash;
+    bytes previousStateMachineState;
```

Comment:
struct BlockEmptyProof {
    SignedBlock emptyBlock;
    SignedBlock previousBlock;
    }
    
    The check is: 
    if(emptyBlock.height == 0 && emptyBlock.previousBlockHash (this is the genesis snapshot hash) == emptyBlock.stateSnapshotHash) -> slash
    if(emptyBlock.previousBlockHash == hash(previousBlock) && emptyBlock.stateSnapshot == previousBlock.stateSnapshot) -> slash
    revert ErrorIncorrectEmptyBlockFraudProof